### ISSUE TITLE
TrivialFamily: Accept start index

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#30259]
<div id="comment:0"></div>

`sage.tensor.modules` has a feature `start_index`. By setting `start_index=1` (default: 0), a basis is indexed starting at 1.

We add this feature to `TrivialFamily` (and the factory `Family`).

This is another small step to improve compatibility with `CombinatorialFreeModule`.

Depends on #30300

CC:  @egourgoulhon @tscrim @mjungmath

Component: **combinatorics**

Branch/Commit: **[u/mkoeppe/trivialfamily__accept_start_index](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/trivialfamily__accept_start_index) @ [`f4f1a60`](https://github.com/sagemath/sagetrac-mirror/commit/f4f1a604683561428af8e40ae7fab62c31ff25c9)**

_Issue created by migration from https://trac.sagemath.org/ticket/30259_

**Comment by jhpalmieri** [comment]
Closing as "won't fix" since that is the branch author's preference.
